File: backend/models/exercise.go
package models

import (
	"time"
)

// Exercise representa un ejercicio
type Exercise struct {
	ID               int      `json:"id" db:"id"`
	Name             string   `json:"name" db:"name"`
	MuscleGroup      string   `json:"muscle_group" db:"muscle_group"`
	PrimaryMuscles   []string `json:"primary_muscles" db:"primary_muscles"`
	SecondaryMuscles []string `json:"secondary_muscles" db:"secondary_muscles"`
	Equipment        string   `json:"equipment" db:"equipment"`
	VideoURL         *string  `json:"video_url" db:"video_url"`
	Bodyweight       bool     `json:"bodyweight" db:"bodyweight"`
	IsSport          bool     `json:"is_sport" db:"is_sport"`
	CreatedAt        time.Time `json:"created_at" db:"created_at"`
}

// ExerciseFilter representa filtros para buscar ejercicios
type ExerciseFilter struct {
	MuscleGroup string `json:"muscle_group"`
	Equipment   string `json:"equipment"`
	Search      string `json:"search"`
	IsSport     *bool  `json:"is_sport"`
}

File: frontend/src/types/exercise.ts
export interface Exercise {
  id: number
  name: string
  muscle_group: string
  primary_muscles: string[]
  secondary_muscles: string[]

  video_url?: string
  bodyweight: boolean
  is_sport: boolean
  created_at: string
}

export interface CreateExerciseRequest {
  name: string
  muscle_group: string
  primary_muscles: string[]
  secondary_muscles: string[]

  video_url?: string
  bodyweight?: boolean
  is_sport?: boolean
}

export interface UpdateExerciseRequest {
  name?: string
  muscle_group?: string
  primary_muscles?: string[]
  secondary_muscles?: string[]

  video_url?: string
  bodyweight?: boolean
  is_sport?: boolean
}

export interface ExerciseFilter {
  muscle_group?: string

  search?: string
  is_sport?: boolean
}
